HW04.Rmd

```diff
@@ -77,45 +175,217 @@ Assume that:
+
+
 (hint maybe looking at percentages would be good)
-```{r race_2}
+```{r race_2, echo=F}
+
+#If there's better way of doing the following I don't know how, but I decomposed the deaths_race data frame into races to calculate the percentages of gun deaths by intent individually.
+black_percent <- deaths_race %>%
+    filter(race == "Black") %>%
+    mutate(percentage = round(deaths / sum(deaths)* 100, digits = 2))
+
+hispanic_percent <- deaths_race %>%
+    filter(race == "Hispanic") %>%
+    mutate(percentage = round(deaths / sum(deaths)* 100, digits = 2))
+
+white_percent <- deaths_race %>%
+    filter(race == "White") %>%
+    mutate(percentage = round(deaths / sum(deaths)* 100, digits = 2))
+
+#Then I merged the data frames per race to gain a new data frame that can be used to make a graph
+deaths_race_percentage <- rbind(black_percent, hispanic_percent, white_percent)
+
+#I initially did this as pie charts, since the percentages allow it, but I kinda like this visualization better, it's much more comparable in my opinion. 
+deaths_race_percentage %>%
+    ggplot(aes(x = race, y = percentage, fill = intent)) +
+    geom_col() +
+    labs(title = "Percentage of gun deaths per race and intent from 2012-2014",
+         x = "Race",
+         y = "Percentage (%)",
+         caption = "Source: fivethirtyeight.com") 
+
+#Finally, a table with the exact data never does harm
+kable(deaths_race_percentage, format = "markdown", col.names = c("Race", "Intent", "Total deaths", "Percentage per race"))
 
 ```
 
 
 ### Are police-involved gun deaths significantly different from other gun deaths? Assess the relationship between police involvement and age, police involvement and race, and the intersection of all three variables.
 
-```{r police}
+```{r police, echo=F}
+deaths_police <- guns %>%
+    select(intent, police, age, race) %>%
+    filter(age != "Not Available", intent == "Homicide")
+
+
+deaths_police$age <- as.integer(deaths_police$age)
+
+#To more understandable variables.
+deaths_police$police <- factor(deaths_police$police)
+levels(deaths_police$police)[levels(deaths_police$police) == 0] <-"Not involved"
+levels(deaths_police$police)[levels(deaths_police$police) == 1] <-"Involved"
+
+#Names here can be too large, so I abbreviate them.
+deaths_police$race <- factor(deaths_police$race)
+levels(deaths_police$race)[levels(deaths_police$race) == "Asian/Pacific Islander"] <-"AP"
+levels(deaths_police$race)[levels(deaths_police$race) == "Black"] <-"BL"
+levels(deaths_police$race)[levels(deaths_police$race) == "Hispanic"] <-"His"
+levels(deaths_police$race)[levels(deaths_police$race) == "Not Availabletive American/Not Availabletive Alaskan"] <-"NNA"
+levels(deaths_police$race)[levels(deaths_police$race) == "White"] <-"WH"
+
+#Police involvement vs. race.
+dp_race <- deaths_police %>%
+    select(police, race) %>%
+    group_by(police) %>%
+    count(race) %>%
+    arrange(race)

```

Comment:
If you arranged by police, it could make comparisons easier across race by having the "involved" and "not involved" stacked up together. Although not asked of you, one last thing to add to this table/data frame would be the relative percentages to help with visualization later down the road

Reply:
Didn't cross my mind, but I saw that Laura Cespedes did it that way, and it looks better. The relative percentages idea is also a very good one.